include/boost/url/impl/url_base.ipp

```diff
@@ -1337,6 +1337,166 @@ end() const noexcept ->
     return iterator(v_, true);
 }
 
+auto
+url_base::
+segments_type::
+erase( url_base::segments_type::iterator pos ) noexcept ->
+    iterator
+{
+    auto last = pos;
+    return erase(pos, ++last);
+}
+
+auto
+url_base::
+segments_type::
+erase( url_base::segments_type::iterator first, url_base::segments_type::iterator last ) noexcept ->
+    iterator
+{
+    BOOST_ASSERT(v_ != nullptr);
+    url_base & v = *v_;
+    BOOST_ASSERT(v.size() == v.a_.size());
+    BOOST_ASSERT(first.v_ == &v);
+    BOOST_ASSERT(last.v_ == &v);
+    BOOST_ASSERT(first.off_ >= v.pt_.offset[detail::id_path]);
+    BOOST_ASSERT(last.off_ >= v.pt_.offset[detail::id_path]);
+    BOOST_ASSERT(first.off_ <= v.pt_.offset[detail::id_query]);
+    BOOST_ASSERT(last.off_ <= v.pt_.offset[detail::id_query]);
+    auto const d = last.off_ - first.off_;
+    if( d == 0 )
+        return first;
+    BOOST_ASSERT(d > 0);
+    int c = 0;
+    for( auto i = v.s_ + first.off_, e = v.s_ + last.off_; i != e; ++i )
+        c += (*i == '/');
```

Comment:
You should put a comment here to explain what the loop does. My guess is that it counts the number of segments between `first` and `last`.

Reply:
Good guess. I'll add the comment.